### ISSUE TITLE
CurrentRefreshRate not working

### ISSUE THREAD

**Issue by mirrorlink** [sebhildebrandt/systeminformation#590]
**Describe the bug**
It seems that the currentRefreshRate is not working, coming as undefined

**To Reproduce**
```
  si.graphics().then((data) =>
  {

    for(let i = 0 ; i < data.controllers.length; i++)
    {
      let control = data.controllers[i];

        console.log(control);
    }

    for(let i = 0 ; i < data.displays.length; i++)
    {
      let display = data.displays[i];

       console.log(display);

    }
  });
```

**Current Output**
undefined

**Expected behavior**
60

**Environment (please complete the following information):**
windows 10
package version 5.9.4

**Additional context**
wmic path win32_VideoController get CurrentRefreshRate

returns the correct values

**Comment by sebhildebrandt** [comment]
@mirrorlink thank you for reporting ... do you have just one controller or more than one? can you provide the full output of 

```
wmic path win32_VideoController get /value
```

I know you already stated that this provides the correct value ... just wanted to be sure to see the full output.
Thank you in advance!

**Comment by mirrorlink** [comment]
> @mirrorlink thank you for reporting ... do you have just one controller or more than one? can you provide the full output of
> 
> ```
> wmic path win32_VideoController get /value
> ```
> 
> I know you already stated that this provides the correct value ... just wanted to be sure to see the full output. Thank you in advance!

No problem! I do have multiple controllers and multiple displays. here it is

```

AcceleratorCapabilities=
AdapterCompatibility=NVIDIA
AdapterDACType=Integrated RAMDAC
AdapterRAM=2147483648
Availability=3
CapabilityDescriptions=
Caption=NVIDIA GeForce GT 1030
ColorTableEntries=
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_VideoController
CurrentBitsPerPixel=32
CurrentHorizontalResolution=1920
CurrentNumberOfColors=4294967296
CurrentNumberOfColumns=0
CurrentNumberOfRows=0
CurrentRefreshRate=60
CurrentScanMode=4
CurrentVerticalResolution=1080
Description=NVIDIA GeForce GT 1030
DeviceID=VideoController1
DeviceSpecificPens=
DitherType=0
DriverDate=20210423000000.000000-000
DriverVersion=27.21.14.6627
ErrorCleared=
ErrorDescription=
ICMIntent=
ICMMethod=
InfFilename=oem48.inf
InfSection=Section061
InstallDate=
InstalledDisplayDrivers=C:\WINDOWS\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_a494df49ba2f9f36\nvldumdx.dll,C:\WINDOWS\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_a494df49ba2f9f36\nvldumdx.dll,C:\WINDOWS\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_a494df49ba2f9f36\nvldumdx.dll,C:\WINDOWS\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_a494df49ba2f9f36\nvldumdx.dll
LastErrorCode=
MaxMemorySupported=
MaxNumberControlled=
MaxRefreshRate=60
MinRefreshRate=50
Monochrome=FALSE
Name=NVIDIA GeForce GT 1030
NumberOfColorPlanes=
NumberOfVideoPages=
PNPDeviceID=PCI\VEN_10DE&amp;DEV_1D01&amp;SUBSYS_11C710DE&amp;REV_A1\4&amp;1286464&amp;0&amp;0008
PowerManagementCapabilities=
PowerManagementSupported=
ProtocolSupported=
ReservedSystemPaletteEntries=
SpecificationVersion=
Status=OK
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=DESKTOP-842KKQT
SystemPaletteEntries=
TimeOfLastReset=
VideoArchitecture=5
VideoMemoryType=2
VideoMode=
VideoModeDescription=1920 x 1080 x 4294967296 colors
VideoProcessor=NVIDIA GeForce GT 1030


AcceleratorCapabilities=
AdapterCompatibility=Intel Corporation
AdapterDACType=Internal
AdapterRAM=1073741824
Availability=3
CapabilityDescriptions=
Caption=Intel(R) HD Graphics 4600
ColorTableEntries=
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_VideoController
CurrentBitsPerPixel=32
CurrentHorizontalResolution=1600
CurrentNumberOfColors=4294967296
CurrentNumberOfColumns=0
CurrentNumberOfRows=0
CurrentRefreshRate=60
CurrentScanMode=4
CurrentVerticalResolution=900
Description=Intel(R) HD Graphics 4600
DeviceID=VideoController2
DeviceSpecificPens=
DitherType=0
DriverDate=20171016000000.000000-000
DriverVersion=20.19.15.4835
ErrorCleared=
ErrorDescription=
ICMIntent=
ICMMethod=
InfFilename=oem40.inf
InfSection=iHSWD_w10
InstallDate=
InstalledDisplayDrivers=igdumdim64.dll,igd10iumd64.dll,igd10iumd64.dll,igd12umd64.dll
LastErrorCode=
MaxMemorySupported=
MaxNumberControlled=
MaxRefreshRate=0
MinRefreshRate=
Monochrome=FALSE
Name=Intel(R) HD Graphics 4600
NumberOfColorPlanes=
NumberOfVideoPages=
PNPDeviceID=PCI\VEN_8086&amp;DEV_0412&amp;SUBSYS_D0001458&amp;REV_06\3&amp;11583659&amp;0&amp;10
PowerManagementCapabilities=
PowerManagementSupported=
ProtocolSupported=
ReservedSystemPaletteEntries=
SpecificationVersion=
Status=OK
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=DESKTOP-842KKQT
SystemPaletteEntries=
TimeOfLastReset=
VideoArchitecture=5
VideoMemoryType=2
VideoMode=
VideoModeDescription=1600 x 900 x 4294967296 colors
VideoProcessor=Intel(R) HD Graphics Family
```

I testes on MACOSX too, and on mac is fine (but i have a single monitor there)

**Comment by sebhildebrandt** [comment]
@mirrorlink thank you so much! will try to find out where the problem is as it should get the refresh rate exactly from this value ... will try to work on it tomorrow when I have access to my windows machines...

**Comment by mirrorlink** [comment]
> @mirrorlink thank you so much!I will try to find out where the problem is as it should get the current refresh rate exactly from this value ... will try to work on it tomorrow when I have access to my windows machines...

Thank you! I'm gonna use it to properly solve the problem with my game, Wirewalk()

https://store.steampowered.com/app/1636700/Wirewalk/

I need to know the FPS as to step Box2D physics accordingly. Currently im measuring it with rAF but it can be so imprecise. I'm afraid maybe it will lead to misleading fps on some people, making the game very slow or very fast.

**Comment by sebhildebrandt** [comment]
@mirrorlink one more thing, can you then provide also full output of:

```
const si = require('systeminformation');
si.graphics().then(data => console.log(JSON.stringify(data, null, 2)));
```

Thank you so much ...

**Comment by sebhildebrandt** [comment]
@mirrorlink ... some thoughts:

- well yes, the currentRefreshRate is something, that WMIC returns on the controller level which is anyway strange because - as in your case - if you have two monitors, this might be no correct information.
- seeing your output I am wondering where the line `"refreshRate": 60,` in the controllers come from ... did you made changes to your local systeminformation code?
- and still I am wondering, why no `currentRefreshRate` shows up in display[0] ... at least in the current code this should work and be set (maybe incorrectly ... see above) to display[0]
- so let me know if you already made some modifications in your local systeminformation code or if this is the original code from GitHub.
- I thought about using `dxdiag` but this requires, that I rely on writing a file to the computer where the code runs and I would consider this as highly risky ... this is the reason why I did not implemented this for now. I did not find any solution to get the output directly to node.js

**Comment by sebhildebrandt** [comment]
@mirrorlink ... ahhh missed the line 847 ;-) ... ok, this was intentionally because having ONE monitor per controller then the value would make kind of sense... in case of more than one, this is not really the correct way. If you find a way to pick up data from dxdiag without writing to a file (even if this is just a sample snippet code) I would be happy if you can share it. I will then make sure to add this to systeminformation.

**Comment by mirrorlink** [comment]
> @mirrorlink cannot find a method to get dxdiag data to node without writing it to disk ... any idea?

Hey! you were correct. I could not find a way either. I've looked for some third party softwares that say they do that but havent testes any.

But all in all, I found it to be too risky. What I end up doing was to use setTimeout instead of requestAnimationFrame for monitors which I detect >64fps (timing requestAnimationFrame). It doesnt look as good but at least It seems more reliable :/
